extensions/ql-vscode/src/interface-types.ts

```diff
@@ -16,6 +16,9 @@ export interface DatabaseInfo {
   databaseUri: string;
 }
 
+/** Arbitrary query metadata */
+export type QueryMetadata = {[key: string]: string};
```

Comment:
I don't really understand the motivation behind the change to this type. What do we gain from *not* expressing some understanding of the structure of the metadata? Right now the only ones we use are `kind` and `id`, but it seems useful as documentation in case we want to use other fields in the future.